### PR TITLE
Return package with already exists

### PR DESCRIPTION
**Pull request by avdata99**
Fixes issue #2

### CODE REVIEW

File: harvester_adapters/ckan/api.py [hunk 1]
```diff
@@ -294,10 +294,14 @@ def create_package(self, ckan_package,
...
-            if json_content['error'].get('name', None) == ["That URL is already in use."]:
-                logger.error(f'Package Already exists! ACTION: {on_duplicated}')
+            dataset_exists = "That URL is already in use" in json_content['error'].get('name', '')
+            harvest_exists = "There already is a Harvest Source for this URL" in json_content['error'].get('url', '')
+            is_duplicated = dataset_exists or harvest_exists
+            if is_duplicated:
+                logger.error(f'Already exists! ACTION: {on_duplicated}')
                 if on_duplicated == 'SKIP':
-                    return {'success': True}
+                    # returns {'success': True, 'result': {the package}}
```
Comment (thejuliekramer) [inline harvester_adapters/ckan/api.py]: Can you remove this comment?

File: harvester_adapters/ckan/api.py [hunk 2]
```diff
@@ -294,10 +294,14 @@ def create_package(self, ckan_package,
...
+                    return self.show_package(ckan_package_id_or_name=ckan_package['name'])
```
Comment (thejuliekramer) [inline harvester_adapters/ckan/api.py]: does this really return {'success': True, 'result': {the package}}? it looks like show package just returns json

File: harvesters/datajson/harvester.py [hunk 3]
```diff
@@ -86,13 +86,15 @@ def validate(self, validator_schema):
...
         self.schema_version = VALID_DATAJSON_SCHEMAS[validator_schema]
         
-        try:
-            self.data_json = json.loads(self.raw_data_json)  # check for encoding errors
-        except Exception as e:
-            error = 'ERROR parsing JSON: {}. Data: {}'.format(e, self.raw_data_json)
-            self.errors.append(error)
-            logger.error(error)
-            return False
+        # check for encoding errors only if we not started this from a dict (as we do in some tests)
```
Comment (thejuliekramer) [inline harvesters/datajson/harvester.py]:
```suggestion
        # check to see if the original json is from a dictionary which will indicate it is a test - we only need to check for encoding errors on real harvests
```

File: harvester_adapters/ckan/api.py (hunk 2 above)
Comment (avdata99) [reply harvester_adapters/ckan/api.py]: show_package return the CKAN API response.
Here an example: https://github.com/datopian/ckan-ng-harvester-core/blob/3_avoid_409_error/tests_with_ckan/test_harvest.py#L78-L80
The response it's a dict. Here [an example](https://catalog.data.gov/api/3/action/package_show?id=8a311c18-3060-438e-a8c7-9e37bcde6529)

**Pull request merged**